sda/protocol.go

```diff
@@ -59,6 +59,10 @@ func ProtocolRegisterName(name string, protocol NewProtocol) ProtocolID {
 		protocols = make(map[ProtocolID]NewProtocol)
 		protocolNames = make(map[ProtocolID]string)
 	}
+	if _, exists := protocolNames[u]; exists {
+		log.Warn("Protocol", u, "already exists - not overwriting")
```

Comment:
I also agree with you but this should be justified and specified by a comment, so we know what to expect from this function (invariants and so on) ... :) 
(Thanks)